bot/exts/moderation/silence.py

```diff
@@ -93,87 +93,222 @@ async def _async_init(self) -> None:
         await self.bot.wait_until_guild_available()
 
         guild = self.bot.get_guild(Guild.id)
-        self._verified_role = guild.get_role(Roles.verified)
+
+        self._verified_msg_role = guild.get_role(Roles.verified)
+        self._verified_voice_role = guild.get_role(Roles.voice_verified)
+        self._helper_role = guild.get_role(Roles.helpers)
+
         self._mod_alerts_channel = self.bot.get_channel(Channels.mod_alerts)
+
         self.notifier = SilenceNotifier(self.bot.get_channel(Channels.mod_log))
         await self._reschedule()
 
+    async def _get_related_text_channel(self, channel: VoiceChannel) -> Optional[TextChannel]:
+        """Returns the text channel related to a voice channel."""
+        # TODO: Figure out a dynamic way of doing this
+        channels = {
+            "off-topic": Channels.voice_chat,
+            "code/help 1": Channels.code_help_voice,
+            "code/help 2": Channels.code_help_voice,
+            "admin": Channels.admins_voice,
+            "staff": Channels.staff_voice
+        }
+        for name in channels.keys():
+            if name in channel.name.lower():
+                return self.bot.get_channel(channels[name])
+
+    async def send_message(self, message: str, source_channel: TextChannel,
+                           target_channel: Union[TextChannel, VoiceChannel],
+                           alert_target: bool = False, duration: HushDurationConverter = 0) -> None:
+        """Helper function to send message confirmation to `source_channel`, and notification to `target_channel`."""
+        await source_channel.send(
+            message.replace("current", target_channel.mention if source_channel != target_channel else "current")
+                   .replace("{duration}", str(duration))
```

Comment:
This will be more readable as a normal if statement rather than an inline one. It also slightly bothers me that in the else case you're redundantly replacing the string with itself.